src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,835 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+        {
+            string referencedTableFullName = sqlMetadataProvider.EntityToDatabaseObject[referencedEntityName].FullName;
+            string referencingTableFullName = sqlMetadataProvider.EntityToDatabaseObject[referencingEntityName].FullName;
+
+            foreach (ForeignKeyDefinition foreignKeyDefinition in foreignKeyDefinitions)
+            {
+                // For a M:N relationship, the referncing table is always the linking table.
+                // So, the referenced table is the only table that needs to be checked.
+                if (isMToNRelationship)
+                {
+                    if (string.Equals(referencedTableFullName, foreignKeyDefinition.Pair.ReferencedDbTable.FullName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return foreignKeyDefinition;
+                    }
+                }
+                else
+                {
+                    if (string.Equals(referencingTableFullName, foreignKeyDefinition.Pair.ReferencingDbTable.FullName, StringComparison.OrdinalIgnoreCase)
+                   && string.Equals(referencedTableFullName, foreignKeyDefinition.Pair.ReferencedDbTable.FullName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return foreignKeyDefinition;
+                    }
+                }
+            }
+
+            throw new DataApiBuilderException(message: $"Foreign Key Definition does not exist with referencing entity: {referencingEntityName} and referenced entity: {referencedEntityName}",
+                                              statusCode: HttpStatusCode.InternalServerError,
+                                              subStatusCode: DataApiBuilderException.SubStatusCodes.UnexpectedError);
+
+        }
+
+        /// <summary>
+        /// Helper method to extract the primary key fields from all the fields of the entity.
+        /// </summary>
+        /// <param name="sqlMetadataProvider">SqlMetadaProvider object for the given database</param>
+        /// <param name="entityName">Name of the entity</param>
+        /// <param name="entityFields">All the fields belonging to the entity</param>
+        /// <returns>Primary Key fields</returns>
+        private static Dictionary<string, object?> FetchPrimaryKeyFields(ISqlMetadataProvider sqlMetadataProvider, string entityName, Dictionary<string, object?> entityFields)
+        {
+            Dictionary<string, object?> pkFields = new();
+            SourceDefinition sourceDefinition = sqlMetadataProvider.GetSourceDefinition(entityName);
+            foreach (string primaryKey in sourceDefinition.PrimaryKey)
+            {
+                if (sqlMetadataProvider.TryGetExposedColumnName(entityName, primaryKey, out string? name))
+                {
+                    if (entityFields.TryGetValue(name, out object? value))
```

Comment:
Are we okay if a PK field has null value? Or do we want to throw an exception in that case?

Reply:
Updated to throw an exception.